### PR TITLE
Fix behavior to display helper function

### LINKED ISSUE #10350: [mesheryctl] `exp model`: Inconsistent error response
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior
```
$ mesheryctl exp model
Error: Meshery server is not reachable: dial tcp: lookup tcp///playground.meshery.io: nodename nor servname provided, or not known
```

### Desired Behavior
This should be a standard response from the set of helper functions.

### Environment

- Host OS: Mac 
- Meshery Server Version: stable-v.7.21
- Meshery Client Version: stable-v0.7.15
- Platform: Docker

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- 📺 [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- ⌨️ [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- ⌨️ [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by lekaf974**
**Notes for Reviewers**

This PR fixes #10350 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.
 

```bash
[lekaf974@ikki mesheryctl]$ ./mesheryctl exp model
View list of models and detailed information of a specific model

Usage:
  mesheryctl exp model [flags]
  mesheryctl exp model [command]

Examples:

// To view list of components
mesheryctl exp model list

// To view a specific model
mesheryctl exp model view [model-name]
	

Available Commands:
  list        list registered models
  search      search models
  view        view model

Flags:
  -h, --help   help for model

Global Flags:
      --config string   path to config file (default "/home/matthieu/.meshery/config.yaml")
  -v, --verbose         verbose output

Use "mesheryctl exp model [command] --help" for more information about a command.
View list of models and detailed information of a specific model

Usage:
  mesheryctl exp model [flags]
  mesheryctl exp model [command]

Examples:

// To view list of components
mesheryctl exp model list

// To view a specific model
mesheryctl exp model view [model-name]
	

Available Commands:
  list        list registered models
  search      search models
  view        view model

Flags:
  -h, --help   help for model

Global Flags:
      --config string   path to config file (default "/home/matthieu/.meshery/config.yaml")
  -v, --verbose         verbose output

Use "mesheryctl exp model [command] --help" for more information about a command.
```

@leecalcote this is the new behavior.

### CODE REVIEW

**Comment by vishalvivekm** [comment]
Thank you @lekaf974 
Let's discuss this on Meshery Dev call, add this as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit) , if you would, please.

**Review by RipulHandoo — COMMENTED** [review]
Could you please clarify why we're logging the command usage twice in the PR description?

File: mesheryctl/internal/cli/root/system/model.go [hunk 1]
```diff
@@ -426,6 +426,13 @@ mesheryctl exp model list
 mesheryctl exp model view [model-name]
 	`,
 	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
```
Comment (RipulHandoo) [inline mesheryctl/internal/cli/root/system/model.go:429]: Could you move the logic to the Args section for clarity?

**Review by RipulHandoo — CHANGES_REQUESTED** [review]

**Comment by lekaf974** [comment]
@RipulHandoo by moving the logic in Args section the helper respponse is displayed following by the error message, is this the expected behavior ? 

new output
```bash
View list of models and detailed information of a specific model

Usage:
  mesheryctl exp model [flags]
  mesheryctl exp model [command]

Examples:

// To view list of components
mesheryctl exp model list

// To view a specific model
mesheryctl exp model view [model-name]
        

Available Commands:
  list        list registered models
  search      search models
  view        view model

Flags:
  -h, --help   help for model

Global Flags:
      --config string   path to config file (default "/home/matthieu/.meshery/config.yaml")
  -v, --verbose         verbose output

Use "mesheryctl exp model [command] --help" for more information about a command.
Error: Meshery server is not reachable: dial tcp [::1]:9081: connect: connection refused

```

changes
image

**Comment by RipulHandoo** [comment]
> @RipulHandoo by moving the logic in Args section the helper respponse is displayed following by the same error message of the opened issue, is this the expected behavior ?
> 
> new output
> 
> ```shell
> View list of models and detailed information of a specific model
> 
> Usage:
>   mesheryctl exp model [flags]
>   mesheryctl exp model [command]
> 
> Examples:
> 
> // To view list of components
> mesheryctl exp model list
> 
> // To view a specific model
> mesheryctl exp model view [model-name]
>         
> 
> Available Commands:
>   list        list registered models
>   search      search models
>   view        view model
> 
> Flags:
>   -h, --help   help for model
> 
> Global Flags:
>       --config string   path to config file (default "/home/matthieu/.meshery/config.yaml")
>   -v, --verbose         verbose output
> 
> Use "mesheryctl exp model [command] --help" for more information about a command.
> Error: Meshery server is not reachable: dial tcp [::1]:9081: connect: connection refused
> ```

Nice.

File: mesheryctl/internal/cli/root/system/model.go [hunk 2]
```diff
@@ -446,6 +446,12 @@ mesheryctl exp model view [model-name]
...
 	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			return cmd.Help()
+		}
+		return nil
+	},
 	RunE: func(cmd *cobra.Command, args []string) error {
 		if len(args) == 0 {
 			return cmd.Help()
```
Comment (RipulHandoo) [inline mesheryctl/internal/cli/root/system/model.go:457]:
```suggestion
	
```

File: mesheryctl/internal/cli/root/system/model.go [hunk 3]
```diff
@@ -446,6 +446,12 @@ mesheryctl exp model view [model-name]
...
 		return nil
 	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			return cmd.Help()
+		}
+		return nil
+	},
 	RunE: func(cmd *cobra.Command, args []string) error {
 		if len(args) == 0 {
```
Comment (RipulHandoo) [inline mesheryctl/internal/cli/root/system/model.go:456]:
```suggestion
```

File: mesheryctl/internal/cli/root/system/model.go [hunk 4]
```diff
@@ -446,6 +446,12 @@ mesheryctl exp model view [model-name]
 		}
 		return nil
 	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			return cmd.Help()
```
Comment (RipulHandoo) [inline mesheryctl/internal/cli/root/system/model.go:451]: @lekaf974 Ensure error handling consistency: Consider returning errors for failed cmd.Help() calls and provide descriptive messages for better debugging.

**Review by RipulHandoo — CHANGES_REQUESTED** [review]

File: mesheryctl/internal/cli/root/system/model.go (hunk 4 above)
Comment (alphaX86) [inline mesheryctl/internal/cli/root/system/model.go:451]:
```suggestion
			return cmd.Help()
```
You can return an error message as @RipulHandoo said. Eg: `This command takes at least one arg.` and then you could use `cmd.Help()` to display the help message

**Review by alphaX86 — CHANGES_REQUESTED** [review]

**Comment by lekaf974** [comment]
@RipulHandoo @alphaX86 ready for review

File: mesheryctl/internal/cli/root/system/model.go [hunk 5]
```diff
@@ -446,10 +446,17 @@ mesheryctl exp model view [model-name]
...
 	},
-	RunE: func(cmd *cobra.Command, args []string) error {
+	Args: func(cmd *cobra.Command, args []string) error {
 		if len(args) == 0 {
-			return cmd.Help()
+			if err := cmd.Usage(); err != nil {
+
+				return nil
+			}
+			return errors.New("please provide a subcommand")
```
Comment (RipulHandoo) [inline mesheryctl/internal/cli/root/system/model.go:455]: Define an ErrInvalidArgs() (generic function).

**Review by RipulHandoo — CHANGES_REQUESTED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit 7ea325a3bc02578eee5e6edd5591265ed52f4d0b at: https://661da26a32900c42e161d248--meshery-docs-preview.netlify.app

File: mesheryctl/internal/cli/root/system/model.go [hunk 6]
```diff
@@ -446,10 +446,16 @@ mesheryctl exp model view [model-name]
 		}
 		return nil
 	},
-	RunE: func(cmd *cobra.Command, args []string) error {
+	Args: func(cmd *cobra.Command, args []string) error {
 		if len(args) == 0 {
-			return cmd.Help()
+			if err := cmd.Usage(); err != nil {
+				return nil
```
Comment (RipulHandoo) [inline mesheryctl/internal/cli/root/system/model.go:452]:
```suggestion
				return err
```

**Review by RipulHandoo — CHANGES_REQUESTED** [review]

**Review by RipulHandoo — APPROVED** [review]

**Comment by lekaf974** [comment]
@alphaX86 approve is still required

**Review by leecalcote — APPROVED** [review]
Looking good.

**Pull request merged**